include/RevOpts.h

```diff
@@ -59,22 +90,24 @@ class RevOpts {
   bool InitPrefetchDepth( const std::vector<std::string>& Depths );
 
   /// RevOpts: retrieve the start address for the target core
-  bool GetStartAddr( uint32_t Core, uint64_t& StartAddr );
+  bool GetStartAddr( uint32_t Core, uint64_t& StartAddr ) const { return GetProperty( Core, startAddr, StartAddr ); }
 
   /// RevOpts: retrieve the start symbol for the target core
-  bool GetStartSymbol( uint32_t Core, std::string& Symbol );
+  bool GetStartSymbol( uint32_t Core, std::string& Symbol ) const { return GetProperty( Core, startSym, Symbol ); }
 
   /// RevOpts: retrieve the machine model string for the target core
-  bool GetMachineModel( uint32_t Core, std::string& MachModel );
+  bool GetMachineModel( uint32_t Core, std::string& MachModel ) const { return GetProperty( Core, machine, MachModel ); }
 
   /// RevOpts: retrieve instruction table for the target core
-  bool GetInstTable( uint32_t Core, std::string& Table );
+  bool GetInstTable( uint32_t Core, std::string& Table ) const { return GetProperty( Core, table, Table ); }
 
   /// RevOpts: retrieve the memory cost range for the target core
-  bool GetMemCost( uint32_t Core, uint32_t& Min, uint32_t& Max );
+  bool GetMemCost( uint32_t Core, uint32_t& Min, uint32_t& Max ) const {
+    return GetProperty( Core, memCosts, std::tie( Min, Max ) );
```

Comment:
`std::tie( Min, Max )` is passed to `GetProperty( )`, whose universal reference `VAL` will be deduced to be an rvalue reference. But it can still be assigned to, and the assignment to `Val` will extract elements of `std::pair` or `std::tuple` into `Min` and Max`.